File: perl/points.pl
#!/usr/bin/env perl
use v5.10;
use strict;
use warnings;
use FFI::Raw;

my $make_point = FFI::Raw->new(
    "target/libpoints.so", "make_point",
    FFI::Raw::ptr,
    FFI::Raw::int,
    FFI::Raw::int
);

my $get_distance = FFI::Raw->new(
    "target/libpoints.so", "get_distance",
    FFI::Raw::double,
    FFI::Raw::ptr,
    FFI::Raw::ptr
);

my $one_point = $make_point->call(2,2);
my $two_point = $make_point->call(4, 4);

my $result = $get_distance->call($one_point, $two_point);

say $result;

File: julia/points.jl
function make_point(a::Int, b::Int)
  ccall( (:make_point, "target/libpoints"), Ptr{Void}, (Int64, Int64), a, b)
end

function get_distance(a::Ptr{Void}, b::Ptr{Void})
  ccall( (:get_distance, "target/libpoints"), Float64, (Ptr{Void}, Ptr{Void}), a, b)
end


t = make_point(2, 2)
u = make_point(4, 4)

v = get_distance(t, u)
println(v)
